### PR TITLE
Remove HTTP_PUSH_ALREADY_IN_CACHE error

### LINKED ISSUE #2812: Rationalise Server-Push-specific rejection error codes
RFC 7540 section 8.2.2 states:

>   If the client determines, for any reason, that it does not wish to
   receive the pushed response from the server or if the server takes
   too long to begin sending the promised response, the client can send
   a RST_STREAM frame, using either the CANCEL or REFUSED_STREAM code
   and referencing the pushed stream's identifier.

HTTP/3 changed push a bit and introduced two error codes: HTTP_PUSH_REFUSED and HTTP_PUSH_ALREADY_IN_CACHE. These is extra fidelity on top of HTTP/2, clearly articulating if the pushed item is in cache or not. This may be seen as useful, however there are some practical downsides:

* The CANCEL_PUSH frame exists and is the first RECOMMENDED action for declining a push. This does not expose the reason
* In actuality, server push caching is murky, which may affect the amount of confidence an operator could put into analyising these error codes.

I think it is simpler, more self-consistent, and more HTTP/2-consistent to have one error code for server push. I'll prepare a PR for this.

### PR DESCRIPTION
**Pull request by LPardue**
Fixes #2812

### CODE REVIEW

File: draft-ietf-quic-http.md [hunk 1]
```diff
@@ -785,10 +785,9 @@ amount of data a server may commit to the pushed stream.
 
 If a promised server push is not needed by the client, the client SHOULD send a
 CANCEL_PUSH frame. If the push stream is already open or opens after sending the
-CANCEL_PUSH frame, a QUIC STOP_SENDING frame with an appropriate error code can
-also be used (e.g., HTTP_PUSH_REFUSED, HTTP_PUSH_ALREADY_IN_CACHE; see
-{{errors}}). This asks the server not to transfer additional data and indicates
-that it will be discarded upon receipt.
+CANCEL_PUSH frame, a QUIC STOP_SENDING frame with an error code of
+HTTP_PUSH_REFUSED. This asks the server not to transfer additional data and
```
Comment (MikeBishop) [inline draft-ietf-quic-http.md]: This sentence is missing a verb.

**Comment by LPardue** [comment]
My intention was to keep PRs somewhat atomic and fragment the error code space in each PR that separately removes or consolidates an error code.

Once complete, I would then defrag it on a PR that let's people weigh in on which error code should have which number.

(It also simplifies dependencies which having branches off branches)

Happy to do something different if you want.

File: draft-ietf-quic-http.md [hunk 2]
```diff
@@ -785,10 +785,9 @@ amount of data a server may commit to the pushed stream.
 
 If a promised server push is not needed by the client, the client SHOULD send a
 CANCEL_PUSH frame. If the push stream is already open or opens after sending the
-CANCEL_PUSH frame, a QUIC STOP_SENDING frame with an appropriate error code can
```
Comment (kazuho) [inline draft-ietf-quic-http.md]: To me the subject and the verb being omitted in this sentence was a bit surprising, but that might be due to me not being a native speaker.

Comment (LPardue) [reply draft-ietf-quic-http.md]: It is an error, thanks for picking it up. I would convert this to

"a QUIC STOP_SENDING frame with an error code of HTTP_REQUEST_CANCELLED can be used."

Alternatively, elsewhere we talk about "triggering QUIC STOP_SENDING" which is under review as part of @MikeBishop #2805

File: draft-ietf-quic-http.md [hunk 3]
```diff
@@ -1903,9 +1901,9 @@ The entries in the following table are registered by this document.
 | ----------------------------------- | ---------- | ---------------------------------------- | ---------------------- |
 | HTTP_NO_ERROR                       | 0x0000     | No error                                 | {{http-error-codes}}   |
 | HTTP_WRONG_SETTING_DIRECTION        | 0x0001     | Setting sent in wrong direction          | {{http-error-codes}}   |
-| HTTP_PUSH_REFUSED                   | 0x0002     | Client refused pushed content            | {{http-error-codes}}   |
+| Reserved                            | 0x0002     | N/A                                      | N/A                    |
```
Comment (martinthomson) [inline draft-ietf-quic-http.md]: My sense is that we want to do a final cleanup for reserved error codes.  We might reserve some for keeping the relationship with RFC 7540 clean, but I don't think that a renumbering is a bad idea after all this work on fixing errors completes.

Comment (LPardue) [reply draft-ietf-quic-http.md]: Definitely. Not many good reasons exist to reserve codes for the shipped RFC.

**Pull request merged**